### PR TITLE
refactor!: outside placement and adjustment for qbx_properties

### PR DESCRIPTION
**Pull request by BerkieBb**
## Description

This PR adds the option in the shared config to allow for outside placement of weed too. It's also adjusted to work with qbx_properties instead of the archived qbx_houses and the SQL has changed to be a bit more understandable and work with the new option. A migration SQL file is provided too.

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

**Comment by BerkieBb** [comment]
The linter test failed because of the source global being only available server side to its knowledge. I submitted a PR to Illenium's linter to fix that

File: config/server.lua [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        min = 1,
+        max = 3
+    },
+    randomHarvestAmount = { -- The random amount of weed to give for a harvest
+        min = 12,
+        max = 16
+    },
+    plantFoodCheckInterval = 1152000, -- How much milliseconds it takes for the plant food to be checked. Default 1152000 milliseconds (19.2 minutes)
+    plantGrowInterval = 576000, -- How much milliseconds it takes for the plant to grow. Default 576000 milliseconds (9.6 minutes)
+    outsidePlantsRefreshInterval = 25000, -- The amount of milliseconds it takes to refresh outside plants. Default 25000 milliseconds (25 seconds)
```
Comment (Manason) [inline config/server.lua]: Would it be easier if these units were made to be seconds or minutes and then converted to milliseconds at runtime?

File: client/main.lua [hunk 2]
```diff
@@ -1,65 +1,121 @@
...
+    if not plantsSpawned or property then return end
+
+    for i = 1, #outsidePlants do
+        local plantData = outsidePlants[i]
+        for stage = 1, #sharedConfig.plants[plantData.sort].stages do
             deleteClosestPlant(stage, plantData)
         end
-        table.remove(plants, i)
+
+        outsidePlants[i] = nil
```
Comment (Manason) [inline client/main.lua]: If each entry in the table is going to be set to nil, would it be easier to set the entire table to empty table '{}' at the conclusion of the for loop?

File: client/main.lua [hunk 3]
```diff
@@ -260,17 +255,114 @@ RegisterNetEvent('qbx_weed:client:foodPlant', function()
...
+        propertyPlants[currentProperty] = plants
+        spawnPropertyPlants()
+    else
+        despawnPropertyPlants(oldValue)
+
+        Wait(1000)
+
+        propertyPlants[oldValue] = nil
+
+        Wait(2000) -- Wait for server to send outsidePlants
```
Comment (Manason) [inline client/main.lua:282]: I'm having a hard time understanding this Wait/comment. Where is the callback or net event that this comment is referring to?

**Review by Manason — APPROVED** [review]
Did a light scan. Looks like a solid improvement

File: config/server.lua (hunk 1 above)
Comment (BerkieBb) [reply config/server.lua]: Yeah, I can do that

File: client/main.lua (hunk 2 above)
Comment (BerkieBb) [reply client/main.lua]: It was doing the same before so I kept it the same but without it using table.remove as that is unnecessary.

File: client/main.lua (hunk 3 above)
Comment (BerkieBb) [reply client/main.lua:282]: It's referring to the event handler on the server onMetadata that calls refreshOutsidePlants

Comment (BerkieBb) [reply client/main.lua:282]: It's too long of a wait, but that's on purpose just in case it can take a little longer

**Review by mafewtm — APPROVED** [review]

**Comment by BerkieBb** [comment]
All comments have been addressed

**Comment by BerkieBb** [comment]
If you run the linter again, it should succeed now, my PR got merged

**Pull request merged**